File: test/benchmark/python/equality.py
import time

i = 0
loopStart = time.process_time()

while i < 10000000:
    i = i + 1

    1
    1
    1
    2
    1
    None
    1
    "str"
    1
    True

    None
    None
    None
    1
    None
    "str"
    None
    True

    True
    True
    True
    1
    True
    False
    True
    "str"
    True
    None

    "str"
    "str"
    "str"
    "stru"
    "str"
    1
    "str"
    None
    "str"
    True

loopTime = time.process_time() - loopStart
start = time.process_time()
i = 0

while i < 10000000:
    i = i + 1

    1 == 1
    1 == 2
    1 == None
    1 == "str"
    1 == True

    None == None
    None == 1
    None == "str"
    None == True

    True == True
    True == 1
    True == False
    True == "str"
    True == None

    "str" == "str"
    "str" == "stru"
    "str" == 1
    "str" == None
    "str" == True

elapsed = time.process_time() - start

print("loop")
print(loopTime)
print("elapsed")
print(elapsed)
print("equals")
print(elapsed - loopTime)

File: test/benchmark/ruby/equality.rb
i = 0
loopStart = Time.now

while i < 10000000
    i = i + 1

    1
    1
    1
    2
    1
    nil
    1
    "str"
    1
    true

    nil
    nil
    nil
    1
    nil
    "str"
    nil
    true

    true
    true
    true
    1
    true
    false
    true
    "str"
    true
    nil

    "str"
    "str"
    "str"
    "stru"
    "str"
    1
    "str"
    nil
    "str"
    true
end

loopTime = Time.now - loopStart
start = Time.now
i = 0

while i < 10000000
    i = i + 1

    1 == 1
    1 == 2
    1 == nil
    1 == "str"
    1 == true

    nil == nil
    nil == 1
    nil == "str"
    nil == true

    true == true
    true == 1
    true == false
    true == "str"
    true == nil

    "str" == "str"
    "str" == "stru"
    "str" == 1
    "str" == nil
    "str" == true
end

elapsed = Time.now - start

puts("loop")
p(loopTime)
puts("elapsed")
p(elapsed)
puts("equals")
p(elapsed - loopTime)
